.github/workflows/main.yml

```diff
@@ -14,72 +14,33 @@ on:
     strategy:
+      fail-fast: false
       matrix:
+        os: ["ubuntu-latest", "macOS-latest"]
         sublime_text:
           - version: 3
             lsp_tag: 'st3'
           - version: 4
             lsp_tag: 'main'
-    env:
-      SUBLIME_TEXT_VERSION: ${{ matrix.sublime_text.version }}
+    runs-on: ${{ matrix.os }}
     steps:
-      - uses: actions/checkout@v1
       - uses: actions/setup-node@v2
         with:
           node-version: '14'
-      - run: echo "$HOME/.local/bin" >> $GITHUB_PATH
-      - run: echo "$HOME/Library/Python/3.7/bin" >> $GITHUB_PATH
-      - run: curl -OL https://raw.githubusercontent.com/SublimeText/UnitTesting/master/sbin/github.sh
-      - run: |
-          sh github.sh bootstrap true  # true == skip package copy
-          sh github.sh install_package LSP ${{ matrix.sublime_text.lsp_tag }} https://github.com/sublimelsp/LSP.git
-          sh github.sh install_package LSP-pyright master https://github.com/sublimelsp/LSP-pyright.git
-          sh github.sh install_package_control
-          sh github.sh copy_tested_package true  # true == overwrite existing package directory
-          sh github.sh run_tests
-
-  Linux:
-    name: Tests (Linux, ST${{ matrix.sublime_text.version }})
-    runs-on: ubuntu-latest
-    container:
-      image: sublimetext/unittesting
-      options: --cap-add=NET_ADMIN
-    strategy:
-      matrix:
-        sublime_text:
-          - version: 3
-            lsp_tag: 'st3'
-          - version: 4
-            lsp_tag: 'main'
-    env:
-      SUBLIME_TEXT_VERSION: ${{ matrix.sublime_text.version }}
-    steps:
-      - uses: actions/checkout@v1
-      - uses: actions/setup-node@v2
+      - uses: actions/checkout@v2
+      - uses: SublimeText/UnitTesting/actions/setup@v1
         with:
-          node-version: '14'
-      - run: sudo apt update
-      - run: sudo apt install --no-install-recommends -y x11-xserver-utils python3-pip
-      - run: sh -e /etc/init.d/xvfb start
-      - run: echo "$HOME/.local/bin" >> $GITHUB_PATH
-      - run: curl -OL https://raw.githubusercontent.com/SublimeText/UnitTesting/master/sbin/github.sh
-      - run: |
-          sh github.sh bootstrap true  # true == skip package copy
-          sh github.sh install_package LSP ${{ matrix.sublime_text.lsp_tag }} https://github.com/sublimelsp/LSP.git
-          sh github.sh install_package LSP-pyright master https://github.com/sublimelsp/LSP-pyright.git
-          sh github.sh install_package_control
-          sh github.sh copy_tested_package true  # true == overwrite existing package directory
-          sh github.sh run_tests
+          sublime-text-version: ${{ matrix.sublime_text.version }}
+          extra-packages: |
+            sublimelsp/LSP@${{ matrix.sublime_text.lsp_tag }}
+            sublimelsp/LSP-pyright
+      - uses: SublimeText/UnitTesting/actions/run-tests@v1
+        with:
+          sublime-text-version: ${{ matrix.sublime_text.version }}
```

Comment:
one last nit. `sublime-text-version` is not needed for `run-test`. It is only needed in the setup step.

Reply:
thanks. also just noticed looking at the output